src/Y2PythonClientComponent.cc

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------\
+|                                                                      |
+|                      __   __    ____ _____ ____                      |
+|                      \ \ / /_ _/ ___|_   _|___ \                     |
+|                       \ V / _` \___ \ | |   __) |                    |
+|                        | | (_| |___) || |  / __/                     |
+|                        |_|\__,_|____/ |_| |_____|                    |
+|                                                                      |
+|                                                                      |
+| python language support                            (C) Novell Inc.   |
+\----------------------------------------------------------------------/
+
+Author: Duncan Mac-Vicar <dmacvicar@suse.de>
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version
+2 of the License, or (at your option) any later version.
+
+*/
+
+#define y2log_component "Y2PythonClient"
+#include <Python.h>
+#include <ycp/y2log.h>
+#include <ycp/pathsearch.h>
+#include <ycp/YCPSymbol.h>
+
+#include "Y2PythonClientComponent.h"
+#include "wfm/Y2WFMComponent.h"
+#include "ycp/YCPBoolean.h"
+
+using std::string;
+
+Y2PythonClientComponent* Y2PythonClientComponent::_instance = NULL;
+
+Y2PythonClientComponent::Y2PythonClientComponent()
+{
+}
+
+Y2PythonClientComponent::~Y2PythonClientComponent()
+{
+  y2debug( "Destroying Y2PythonClientComponent" );
+}
+
+Y2PythonClientComponent* Y2PythonClientComponent::instance()
+{
+  if (_instance == NULL)
+    _instance = new Y2PythonClientComponent();
+
+  return _instance;
+}
+
+YCPValue callClient(const string& client)
+{
+    FILE *fp = fopen(client.c_str(), "r");
+    int res = 0;
+    char *arg1 = (char*)client.c_str();
+    if (fp == NULL) {
+        return YCPBoolean(false);
+    }
+
+    Py_Initialize();
+    PySys_SetArgv(1, &arg1);
+    res = PyRun_SimpleFile(fp, client.c_str());
+    Py_Finalize();
+    fclose(fp);
+    if (res < 0) {
+        return YCPBoolean(false);
+    }
+    return YCPBoolean(true);
```

Comment:
what about `return YCPBoolean(res < 0)` ?

Reply:
I pulled that straight from the logic in the old bindings. Didn't really consider simplifying it :)

Reply:
Oh, I lied. Noel wrote this.

Reply:
I think the logic would be:
return YCPBoolean(res >= 0);
Since res < 0 == false
YCPBoolean(res < 0) would return the opposite logic.